launchdarkly-server-sdk/src/events/sender.rs

```diff
@@ -107,6 +122,21 @@ where
                 }
             };
 
+            // mut is needed for event-compression feature
+            #[allow(unused_mut)]
+            let mut additional_headers = self.default_headers.clone();
```

Comment:
Might be not worth it in terms of readability, but possibly this could conditionally be `mut` if event-compression is enabled, and not mut otherwise.

Reply:
I have to do the same thing with the json bit above, so the readability doesn't seem good. I'm inclined to leave it.